articles/building-a-simple-cryptocurrency-blockchain/index.md

```diff
@@ -0,0 +1,187 @@
+
+1. [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
+2. [Node.js](https://nodejs.org/en/docs/)
+
+## Prerequisite
+To follow this tutorial successfully, you’ll need to have the following:
+
+- Node.js Installed in your machine
+
+- Code editor for my case I use VSCode
+
+
+### What is Blockchain?
+![Bitcoin Image](engineering-education/building-a-simple-cryptocurrency-blockchain/bitcoin.jpg)
+Digital currencies such as Bitcoin, Ethereum are powered and adopted with powerful technology **the Blockchain**. It uses Cryptography to securely connect and maintain a list of records growing continuously known as **Blocks**
+![ethereum](engineering-education/building-a-simple-cryptocurrency-blockchain/ethereum.jpg)
+**Blockchain** as the name states it's a block of transactions data growing increasingly to create a chain of transaction occurrences. Valid transaction data are logged into the blockchain network following the peer-to-peer rule laid down by participants.
+
+We'll have good understanding of blockchain and cryptocurrency and it's operation. 
+
+> Let's get into code
+
+> I'll name my app thecoin.
+
+> Create the app named thecoin.js and open it in your code editor
+
+> In the folder you're doing current development let's install the crypto library we're going to use using the command
+
+```bash
+npm install --save crypto-js
+```
+
+We'll use this library to import modules in our project.
+
+**Begin**
+
+I'll begin by creating a class 'BlockCypto'.
+
+```Node
+    const SHA256 = require('crypto-js/sha256');
+    class BlockCypto{
+        constructor(index, current_time, info, nextHash=" "){
+        this.index = index;
+        this.current_time = current_time;
+        this.info = info;
+        this.nextHash = nextHash;
+        this.hash = this.computeHash();     
+        }
+        computeHash(){
+            return SHA256(this.info + this.nextHash + this.current_time + JSON.stringify(this.info)).toString();
+        }   
+    }
+```js
+
+I'll explain each part of the code here:
+
+I've created the class BlockCrytpo for my ***block*** and added a constructor just like any other JavaScript class. 
+
+In the constructor we initialize its properties and Assign parameters to it as follows:
+
+    
+
+- `crypto-js/sha256`: This is the module we've imported to calculate the hash of each block and we convert it to string using 'toString' method as the module will return the object.
+
+- `index`:  This is a distinctive number tracking the index of every block in the blockchain
+
+- `current_time`: As the name states, it keeps a record of the time when each transaction is completed. 
+
+- `info`: Keeps record and provide data about completed transactions 
+
+- `nexthash` : It points to the hash of the next block in the chain, used to keep and maintain the integrity of the blockchain.
+
+- `computeHash` We use this method to calculate the hash of the next block based on its properties.
```

Comment:
Why are we doing this? What happens? Explain briefly

Reply:
> Why are we doing this? What happens? Explain briefly

Hi @srishilesh I've just made explanation of the code below, Are you able to understand it?

Reply:
Hey,

I am able to understand it. But, the reader might not. That's why I suggested.